superset/assets/visualizations/table.js

```diff
@@ -90,10 +90,11 @@ function tableVis(slice, payload) {
         html = `<span class="like-pre">${val}</span>`;
       }
       if (isMetric) {
-        html = slice.d3format(c, val);
+        const numberFormat = slice.datasource.column_formats && slice.datasource.column_formats[c] || fd.number_format || '.3s';
```

Comment:
I should have gotten this on the first pass but I think the right thing to do is to add a new optional arg to `slice.d3format` called `fallbackFormat` or something like that. 

We should also add a comment to that function making it clear what the precedence rule is (column-defined format if available, `fallbackFormat` if it's not, and '0.3s' if no fallback is specified.)